src/components/action-popover/action-popover.test.js

```diff
@@ -0,0 +1,702 @@
+                  onClick={() => props.onClick("Download")}
+                  icon="download"
+                  href="example-img.jpg"
+                >
+                  Download
+                </ActionPopoverItem>
+              </ActionPopover>
+            </FlatTableCell>
+          </FlatTableRow>
+          <FlatTableRow>
+            <FlatTableCell>Bob</FlatTableCell>
+            <FlatTableCell>Jones</FlatTableCell>
+            <FlatTableCell>
+              <ActionPopover>
+                <ActionPopoverItem
+                  icon="csv"
+                  onClick={() => props.onClick("Download CSV")}
+                  submenu={submenuWithIcons}
+                >
+                  Download CSV
+                </ActionPopoverItem>
+              </ActionPopover>
+            </FlatTableCell>
+          </FlatTableRow>
+        </FlatTableBody>
+      </FlatTable>
+    </div>
+  );
+};
+
+const ActionPopoverWithProps = ({ ...props }) => {
+  return (
+    <div
+      style={{
+        height: "250px",
+      }}
+    >
+      <FlatTable>
+        <FlatTableHead>
+          <FlatTableRow>
+            <FlatTableHeader>First Name</FlatTableHeader>
+            <FlatTableHeader>Last Name</FlatTableHeader>
+            <FlatTableHeader>&nbsp;</FlatTableHeader>
+          </FlatTableRow>
+        </FlatTableHead>
+        <FlatTableBody>
+          <FlatTableRow>
+            <FlatTableCell>John</FlatTableCell>
+            <FlatTableCell>Doe</FlatTableCell>
+            <FlatTableCell>
+              <ActionPopover {...props}>
+                <ActionPopoverItem icon="email" disabled onClick={() => {}}>
+                  Email Invoice
+                </ActionPopoverItem>
+                <ActionPopoverDivider />
+                <ActionPopoverItem onClick={() => {}} icon="delete">
+                  Delete
+                </ActionPopoverItem>
+              </ActionPopover>
+            </FlatTableCell>
+          </FlatTableRow>
+        </FlatTableBody>
+      </FlatTable>
+    </div>
+  );
+};
+
+const ActionPopoverMenuWithProps = ({ ...props }) => {
+  return (
+    <ActionPopover>
+      <ActionPopoverMenu {...props}>
+        <ActionPopoverItem>Sub Menu 1</ActionPopoverItem>
+        <ActionPopoverItem>Sub Menu 2</ActionPopoverItem>
+      </ActionPopoverMenu>
+    </ActionPopover>
+  );
+};
+
+context("Test for ActionPopover component", () => {
+  describe("check functionality for ActionPopover component", () => {
+    it("should render ActionPopover component", () => {
+      CypressMountWithProviders(<ActionPopoverCustom />);
+
+      actionPopoverButton().eq(0).click();
+      actionPopover().should("be.visible");
+    });
+
+    it.each([
+      [0, "Business"],
+      [1, "Email Invoice"],
+      [2, "Print Invoice"],
+      [3, "Download PDF"],
+      [4, "Download CSV"],
+      [5, "Delete"],
+    ])(
+      "should render ActionPopover and be able to press Tab %s times and get button %s focused",
```

Comment:
Should read "be able to press down arrow %s times" not Tab as Tab would take the user out of the extended popover.

Reply:
updated